index/src/lib.rs

```diff
@@ -1,35 +1,27 @@
-use std::fmt;
-use std::fmt::Formatter;
+use thiserror::Error;
 
 pub mod index;
 
-#[derive(Debug)]
+#[derive(Debug, Error)]
 pub enum LittIndexError {
+    #[error("Index Creation Error: `{0}`")]
     CreationError(String),
+    #[error("Error updating the index: `{0}`")]
     UpdateError(String),
+    #[error("Error opening existing index: `{0}`")]
     OpenError(String),
+    #[error("Error reloading index writer: `{0}`")]
     ReloadError(String),
+    #[error("Index Write Error: `{0}`")]
     WriteError(String),
+    #[error("Index is not in assumed state: `{0}`")]
     StateError(String),
+    #[error("Index Read Error: `{0}`")]
     ReadError(String),
+    #[error("Error parsing PDF: `{0}`")]
     PdfParseError(String),
+    #[error("Error parsing txt-file: `{0}`")]
     TxtParseError(String),
 }
 
```

Comment:
Love it, however, am I seeing it correctly, that the update has not been added to the search sofar?

Reply:
I think you looked at an old commit, check this:
https://github.com/georgbuechner/litt/blob/d250beea884c711f5b43e586c0b83750af4ca7d6/search/src/lib.rs#L5-L11